arrow-libs/optics/arrow-optics-ksp-plugin/src/main/kotlin/arrow/optics/plugin/internals/processor.kt

```diff
@@ -69,12 +66,13 @@ internal fun evalAnnotatedPrismElement(
 ): List<Focus> =
   when {
     element.isSealed ->
-      element.sealedSubclassFqNameList().map {
+      element.getSealedSubclasses().map {
         Focus(
-          it,
-          it.substringAfterLast(".").replaceFirstChar { c -> c.lowercase(Locale.getDefault()) }
+          it.primaryConstructor?.returnType?.resolve()?.qualifiedString() ?: it.qualifiedNameOrSimpleName,
+          it.simpleName.asString().replaceFirstChar { c -> c.lowercase(Locale.getDefault()) },
+          it.superTypes.first().resolve()
         )
-      }
+      }.toList()
```

Comment:
Does `element.getSealedSubclasses()` return a `Sequence`?

Reply:
It does. Since this is a small sequence anyway, I went the easy route and just turned it into a `List` at the end.